db/db_impl_files.cc

```diff
@@ -99,10 +100,10 @@ void DBImpl::FindObsoleteFiles(JobContext* job_context, bool force,
   if (doing_the_full_scan) {
     InfoLogPrefix info_log_prefix(!immutable_db_options_.db_log_dir.empty(),
         dbname_);
-    std::vector<std::string> paths;
+    std::set<std::string> paths;
     for (size_t path_id = 0; path_id < immutable_db_options_.db_paths.size();
          path_id++) {
-      paths.emplace_back(immutable_db_options_.db_paths[path_id].path);
+      paths.insert(immutable_db_options_.db_paths[path_id].path);
```

Comment:
do we still need to clean db_paths now that cf_paths is used everywhere?

Reply:
I have no idea. A user complained we are making too many file listing calls, so I investigate. We are still making a lot of duplicated file listing calls in DB::Open(). This can be a project for open-source community.